pkg/asset/installconfig/azure/session.go

```diff
@@ -128,18 +132,27 @@ func getCredentials(fs auth.FileSettings) (*Credentials, error) {
 		return nil, errors.New("could not retrieve clientId from auth file")
 	}
 	clientSecret := fs.Values[auth.ClientSecret]
-	if clientSecret == "" {
-		return nil, errors.New("could not retrieve clientSecret from auth file")
-	}
 	tenantID := fs.Values[auth.TenantID]
 	if tenantID == "" {
 		return nil, errors.New("could not retrieve tenantId from auth file")
 	}
+	clientCertificatePassword := fs.Values[auth.CertificatePassword]
+	clientCertificatePath := fs.Values[auth.CertificatePath]
+	if tenantID == "" {
+		return nil, errors.New("could not retrieve tenantId from auth file")
+	}
+	if clientSecret == "" && (clientCertificatePassword == "" && clientCertificatePath == "") {
+		return nil, errors.New("could not retrieve either client secret or client certs from auth file")
+	} else if clientCertificatePath != "" {
```

Comment:
Is it possible to have `clientSecret` and only one of `certPath / certPassword` at this point, and should this be validated away with an error?

Reply:
Fixed the if conditions. The installer/terraform needs at least one of the authentication methods for it to contact the APIs on azure so we need to return an error.